app/modules/individuals/schemas.py

```diff
@@ -43,3 +43,14 @@ class Meta(BaseIndividualSchema.Meta):
             Individual.created.key,
             Individual.updated.key,
         )
+
+
+class SightingApiIndividualSchema(BaseIndividualSchema):
+    hasView = base_fields.Function(Individual.current_user_has_view_permission)
```

Comment:
hasView and hasEdit (implementations) should really be moved into the BaseIndividualSchema, as they are in BaseEncounterSchema and then can remove this duplication in the derived Schema classes.